OpenRA.Mods.Common/Traits/Upgrades/UpgradeManager.cs

```diff
@@ -67,20 +74,17 @@ class UpgradeState
 		}
 
 		readonly List<TimedUpgrade> timedUpgrades = new List<TimedUpgrade>();
-		readonly Lazy<Dictionary<string, UpgradeState>> upgrades;
 		readonly Dictionary<IUpgradable, int> levels = new Dictionary<IUpgradable, int>();
+		Dictionary<string, UpgradeState> upgrades;
```

Comment:
I wanted this to throw if the methods are accessed before `Created` was called. I've added an explicit exception with a message since this will be clearer in the code and for consumers.